www/pfSense-pkg-squid/files/usr/local/pkg/squid.xml

```diff
@@ -686,10 +723,58 @@
 			<rows>10</rows>
 			<advancedfield/>
 		</field>
+		<field>
+			<fielddescr>Custom Options (SSL/MITM)</fielddescr>
+			<fieldname>custom_options3_squid3</fieldname>
+			<description>
+				<![CDATA[
+				Put your own custom options here, one per line. They'll be added to the configuration in place of the default SSL/MITM configuration.<br/>
+				<span class="text-info">Click Info for details.</span> 
+				<div class="infoblock">
+				Some predefined ACLs for step values are available here: <strong><span class="text-info">step1, step2, step3</span></strong>.<br/>
+				These can be used there for custom configuration and are applied at the corresponding ssl_bump evaluation moments:<br/><br/>
+				<dl class="dl-horizontal responsive">
+				<dt>step1</dt><dd>SslBump1 - After getting TCP-level and HTTP CONNECT info.</dd>
+				<dt>step2</dt><dd>SslBump2 - After getting TLS Client Hello info.</dd>
+				<dt>step3</dt><dd>SslBump3 - After getting TLS Server Hello info.</dd>
+				</dl>
+				<strong><span class="text-info">Example:</span></strong><br/>
+				<code>
+				# some banking sites that should not be MITM-ed<br/>
+				acl serverIsBank ssl::server_name .bank1.example.com<br/>
+				acl serverIsBank ssl::server_name .bank2.example.net<br/>
+				# some sites we want to monitor/do content filtering for<br/>
+				acl monitoredSites ssl::server_name .foo.example.com<br/>
+				acl monitoredSites ssl::server_name .bar.example.org<br/>
+				# get SNI obtained by parsing TLS Client Hello during step2<br/>
+				# (which is instructed by ssl_bump peek step1)<br/>
+				ssl_bump peek step1<br/>
+				# bump monitored sites, but not banks<br/>
+				ssl_bump bump monitoredSites !serverIsBank<br/>
+				# splice all the rest<br/>
+				ssl_bump splice all
+				</code><br/><br/>
+				Please see <a href="http://wiki.squid-cache.org/Features/SslPeekAndSplice" target="_blank">SslBump Peek and Splice wiki documentation</a>
+				for additional details.
+				</div><br/>
+				<strong><span class="text-danger">Warning:</span> These need to be squid.conf native options, otherwise Squid will NOT work.</strong>
+				]]>
+			</description>
+			<type>textarea</type>
+			<encoding>base64</encoding>
+			<cols>78</cols>
+			<rows>10</rows>
+			<advancedfield/>
+		</field>
 	</fields>
 	<custom_php_validation_command>
 		squid_validate_general($_POST, $input_errors);
 	</custom_php_validation_command>
+	<!--
+	<custom_php_after_form_command>
+		squid_print_javascript_general2();
+	</custom_php_after_form_command>
+	-->
```

Comment:
Well the idea is getting some JS magic there as in Antivirus/Authentication tab, but it's
- not ready yet
- mostly unrelated to this change
- not compatible with RELENG_2_3_2 in case someone decides to cherrypick this to that branch